zstacklib/zstacklib/utils/upload_task.py

```
@@ -220,35 +328,53 @@ def get_upload_task(self, param):
 
     def upload_slice(self, entity, param, task):
         boundary = self.get_boundary(entity)
         if not boundary:
-            self._fail_task(task, 'unexpected post form')
+            err = 'unexpected post form'
+            task.handle_upload_slice_error(param.slice_offset, param.slice_size, err)
+            raise Exception(err)
 
         try:
             self.stream_body(entity, boundary, param, task)
+        except RetryableUploadError, e:
```

Ruff (0.15.17)

[error] Undefined name `e`

(F821)